other_enrichment_methods/nbconverted/3_viz_pathway_ranking.py

```diff
@@ -0,0 +1,223 @@
+camera_pathway_summary = pd.read_csv(camera_pathway_summary_filename, sep="\t", index_col=0, header=0)
+ora_pathway_summary = pd.read_csv(ora_pathway_summary_filename, sep="\t", index_col=0, header=0)
+
+
+# ## Format data for plotting
+
+# In[7]:
+
+
+gsea_pathway_summary.head()
+
+
+# In[8]:
+
+
+gsva_pathway_summary.head()
+
+
+# In[9]:
+
+
+roast_pathway_summary.head()
+
+
+# In[10]:
+
+
+camera_pathway_summary.head()
+
+
+# In[11]:
+
+
+ora_pathway_summary.head()
+
+
+# ## Plot
+
+# In[12]:
+
+
+# define plotting function
+def plot_significance_vs_ranking(summary_df, method_name, x_label, output_figure_filename):
+    # Format input dataframe
+    plot_df = pd.DataFrame(
+        data={"Test statistic": summary_df[method_stats_dict[method_name]+" (Real)"].values,
+          "Percentile rank": summary_df["Rank (simulated)"].rank(pct=True).values
+         },
+        index=summary_df.index
+    )
+    
+    fig = pn.ggplot(plot_df, pn.aes(x='Test statistic', y='Percentile rank'))
+    fig += pn.geom_point()
+    fig += pn.geom_point(plot_df[plot_df['Percentile rank']>0.9],
+                         pn.aes(x='Test statistic', y='Percentile rank'),
+                         color='red'
+                        )
+    fig += pn.geom_text(pn.aes(label=
+                               [x if plot_df.loc[x,'Percentile rank']>0.9 else "" for x in plot_df.index]),
+                        ha='left',
+                        va='top',
+                        size=5
+                       )
+    fig += pn.labs(x = x_label,
+                y = 'Percentile of ranking',
+                title = f'{method_name} pathway statistics vs ranking')
+    fig += pn.theme_bw()
+    fig += pn.theme(
+        legend_title_align = "center",
+        plot_background=pn.element_rect(fill='white'),
+        legend_key=pn.element_rect(fill='white', colour='white'), 
+        legend_title=pn.element_text(family='sans-serif', size=15),
+        legend_text=pn.element_text(family='sans-serif', size=12),
+        plot_title=pn.element_text(family='sans-serif', size=15),
+        axis_text=pn.element_text(family='sans-serif', size=12),
+        axis_title=pn.element_text(family='sans-serif', size=15)
+        )
+
+    print(fig)
+    
+    # Save figure
+    fig.save(
+        output_figure_filename,
+        format="svg",
+        bbox_inches="tight",
+        transparent=True,
+        pad_inches=0,
+        dpi=300,
+    )
+
+
+# In[13]:
+
+
+plot_significance_vs_ranking(gsea_pathway_summary, "GSEA", "adjusted p-value (BH)", "GSEA_pathway_ranking.svg")
+
+
+# In[14]:
+
+
+plot_significance_vs_ranking(gsva_pathway_summary, "GSVA", "Enrichment score", "GSVA_pathway_ranking.svg")
+
+
+# In[15]:
+
+
+plot_significance_vs_ranking(roast_pathway_summary, "ROAST", "FDR (BH)", "ROAST_pathway_ranking.svg")
+
+
+# In[16]:
+
+
+plot_significance_vs_ranking(camera_pathway_summary, "CAMERA", "FDR (BH)", "CAMERA_pathway_ranking.svg")
```

Comment:
These look like they're in alphabetical order (i.e. "adipogenesis", "allograft rejection", "androgen response", etc). Is that just because there are a lot of significant pathways/gene sets (so it's just displaying the first few in alphabetical order), or does the order that gene sets are tested in matter for CAMERA?

Reply:
Looks like by default the table is ordered by the `Rank (Real)` column, which is the rank  of the pathway (based on the enrichment) in the template experiment. I don't think the order should matter as each of the gene set tests are performed independently from what i can tell